Encore tests failing due to float precision

I'm mostly just going to steal @jbarnoud 's debugging work here, so we can talk about the failures in their own issue...

The value error seems unrelated to the numpy version, indeed. analysis.encore.cutils.PureRMSD expects float64_t coordinates but receives float ones. The first commit to fail the test is e356604 (see travis: https://travis-ci.org/MDAnalysis/mdanalysis/builds/192119548). The pull request (#1132) was passing the tests, but it got merged right after #1145 in 789a96c. The two pull requests are touching parts of ENCORE. My guess is that the MemoryReader is now working with float32 rather than float64 which breaks analysis.encore.cutils.PureRMSD. The simple solution would be to change analysis.encore.cutils.PureRMSD to use float32 instead of float64 but I would like to know what @kain88-de and @wouterboomsma think about it before changing this.

And my 2c, is that regardless of what `MemoryReader` is storing/supplying, there should probably be a check and/or `np.asarray` call somewhere to avoid this.

I investigated the problem a bit. [`analysis.encore.cutils.PureRMSD`](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/analysis/encore/cutils.pyx#L41) is what complains, but it is called by [`analysis.encore.condistmatrix.set_rmsd_matrix_elements`](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/analysis/encore/confdistmatrix.py#L173) that has two branches.

The first branch is calling `PureRMSD` without paying attention to the type of the coordinates, this branch is visited when there are not fit coordinates provided, and it is the branch that currently causes the error.

The second branch is visited when there are fir coordinates provided. There the coordinates can be float32 or float64 but they are explicitly casted to float64 before they are fed to `PureRMSD`.

Having everything homogenized to float32 (with explicit cast to float32 in both branched, and `PureRMSD` modified) does not generate errors anymore, but one test fails:

```
======================================================================
FAIL: test_clustering_three_ensembles_two_identical (MDAnalysisTests.analysis.test_encore.TestEncoreClustering)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jon/Envs/oldnp/local/lib/python2.7/site-packages/MDAnalysisTests/analysis/test_encore.py", line 457, in test_clustering_three_ensembles_two_identical
    err_msg="Unexpected result: {0}".format(cluster_collection))
  File "/home/jon/Envs/oldnp/local/lib/python2.7/site-packages/numpy/testing/utils.py", line 381, in assert_equal
    raise AssertionError(msg)
AssertionError: 
Items are not equal:
Unexpected result: 0 (size:2,centroid:2): array([ 2, 43])
1 (size:2,centroid:3): array([ 3, 44])
2 (size:1,centroid:4): array([4])
3 (size:1,centroid:5): array([5])
4 (size:1,centroid:6): array([6])
5 (size:1,centroid:7): array([7])
6 (size:1,centroid:8): array([8])
7 (size:1,centroid:9): array([9])
8 (size:1,centroid:10): array([10])
9 (size:2,centroid:11): array([11, 52])
10 (size:2,centroid:12): array([12, 53])
11 (size:1,centroid:13): array([13])
12 (size:1,centroid:14): array([14])
13 (size:2,centroid:15): array([15, 56])
14 (size:2,centroid:16): array([16, 57])
15 (size:1,centroid:17): array([17])
16 (size:3,centroid:19): array([19, 40, 60])
17 (size:3,centroid:23): array([22, 23, 24])
18 (size:2,centroid:26): array([25, 26])
19 (size:2,centroid:28): array([27, 28])
20 (size:3,centroid:30): array([29, 30, 31])
21 (size:3,centroid:33): array([32, 33, 34])
22 (size:3,centroid:36): array([35, 36, 37])
23 (size:3,centroid:41): array([ 0, 20, 41])
24 (size:3,centroid:42): array([ 1, 21, 42])
25 (size:1,centroid:45): array([45])
26 (size:1,centroid:46): array([46])
27 (size:1,centroid:47): array([47])
28 (size:1,centroid:48): array([48])
29 (size:1,centroid:49): array([49])
30 (size:1,centroid:50): array([50])
31 (size:1,centroid:51): array([51])
32 (size:1,centroid:54): array([54])
33 (size:1,centroid:55): array([55])
34 (size:1,centroid:58): array([58])
35 (size:4,centroid:59): array([18, 38, 39, 59])

 ACTUAL: 36
 DESIRED: 40
```

Everything homogenized to float64 has every tests passing.

I did not try the fused types, and I won't since @mtiberti stepped in :+1:

--- a/package/MDAnalysis/analysis/encore/confdistmatrix.py
+++ b/package/MDAnalysis/analysis/encore/confdistmatrix.py
@@ -209,8 +209,8 @@ def set_rmsd_matrix_elements(tasks, coords, rmsdmat, masses, fit_coords=None,
 
     if fit_coords is None and fit_masses is None:
         summasses = np.sum(masses)
-        rmsdmat[(i + 1) * i / 2 + j] = PureRMSD(coords[i],
-                                                coords[j],
+        rmsdmat[(i + 1) * i / 2 + j] = PureRMSD(coords[i].astype(np.float64),
+                                                coords[j].astype(np.float64),
                                                 coords[j].shape[0],
                                                 masses,
                                                 summasses)

